.github/workflows/release.yml

```diff
@@ -22,6 +22,11 @@ jobs:
       - name: Install Rust toolchain
         uses: dtolnay/rust-toolchain@nightly
 
+      - name: Update version
+        run: |
+          export VERSION=${{ github.event.inputs.release_tag }}
+          sed -i "s/0.0.0/$VERSION/g" Cargo.toml
+
```

Comment:
This is where the magic happens. It overwrites the release version with the one tagged by the pipeline.